src/libexec/zmrcd

```diff
@@ -59,9 +59,11 @@ my %SIMPLE_COMMANDS = (
    "downloadcsr"      => "cat /opt/zimbra/ssl/zimbra/commercial/commercial.csr"
 );
 
+my $REGEX_POSTFIX_QID = qr{(?:[A-F0-9]{6,}|[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-z]{10}z[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-y]+)};
+
 my %REGEX_CHECKED_COMMANDS = (
   "zmqaction" => {
-    regex => '(hold|release|requeue|delete)\s+(incoming|deferred|corrupt|active|maildrop|hold)\s+[A-FL0-9,]+', # A-F allows hex, L allows ALL
+    regex => qr{(hold|release|requeue|delete)\s+(incoming|deferred|corrupt|active|maildrop|hold)\s+(?:(?:${REGEX_POSTFIX_QID},)*+${REGEX_POSTFIX_QID}|ALL)},
```

Comment:
Avoid unnecessary repetition:
```suggestion
    regex => qr{(hold|release|requeue|delete)\s+(incoming|deferred|corrupt|active|maildrop|hold)\s+(?:(?:${REGEX_POSTFIX_QID},?)+|ALL)},
```

Reply:
You alteration fails in my tests, although I haven't had the time to check why yet. And unless I am mistaken it would also match two or more QIDs concatenated without a comma separating them. But I haven't verified yet.

Reply:
@plobbes Your updated version works fine and avoids matching concatenated QIDs, so all the better :) .
Only I would use the possessive quantifier `++` since that way a lot of backtracking is avoided in case of invalid input (verified with a test).
So
```perl
regex => qr{(hold|release|requeue|delete)\s+(incoming|deferred|corrupt|active|maildrop|hold)\s+(?:(?:${REGEX_POSTFIX_QID}(?:,|\Z))++|ALL)},
```
with a `(?:${REGEX_POSTFIX_QID}(?:,|\Z))++` with `++` instead of just `+` .

Reply:
Hello @plobbes. Putting together the modifications now.
I'll have to go back to the original pattern
```perl
(?:(?:${REGEX_POSTFIX_QID},)*+${REGEX_POSTFIX_QID}|ALL)
```
since the simplification
```perl
(?:(?:${REGEX_POSTFIX_QID}(?:,|\Z))++|ALL)
```
matches also something like
```
zmqaction hold deferred 5200B21EE8,42gBt96vDXz8sXf,E901121F6F,
```
with a trailing `,` which would break things.